### ISSUE TITLE
Unable to 'unallow' a user that has been deleted

### ISSUE THREAD

**Issue by darkpixel** [openzfs/zfs#11538]
### System information
Type | Version/Name
 --- | ---
Distribution Name	| Debian
Distribution Version	| 10
Linux Kernel	| 4.19.0-12
Architecture	| amd64
ZFS Version	| 0.8.4-2~bpo10+1
SPL Version	| 0.8.4-2~bpo10+1

### Describe the problem you're observing
There doesn't appear to be a way to 'unallow' a user that no longer exists on the system.

### Describe how to reproduce the problem
Add a test user to the system, create a test folder, and grant that user create permissions on the test folder:
```
root@uswuxsdbkp00:~# useradd test
root@uswuxsdbkp00:~# zfs create tank/myfolder
root@uswuxsdbkp00:~# zfs allow -u test create tank/myfolder
root@uswuxsdbkp00:~# zfs allow tank/myfolder
---- Permissions on tank/myfolder ------------------------------------
Local+Descendent permissions:
	user test create
root@uswuxsdbkp00:~# 
```

Unallow that user to show permissions can be removed:
```
root@uswuxsdbkp00:~# zfs unallow -u test tank/myfolder
root@uswuxsdbkp00:~# zfs allow tank/myfolder
root@uswuxsdbkp00:~# 
```

Re-add the create permission and verify it exists:
```
root@uswuxsdbkp00:~# zfs allow -u test create tank/myfolder
root@uswuxsdbkp00:~# zfs allow tank/myfolder
---- Permissions on tank/myfolder ------------------------------------
Local+Descendent permissions:
	user test create
root@uswuxsdbkp00:~# 
```

Remove the test user from the system and look at the permissions again:
```
root@uswuxsdbkp00:~# userdel test
root@uswuxsdbkp00:~# zfs allow tank/myfolder
---- Permissions on tank/myfolder ------------------------------------
Local+Descendent permissions:
	user  create
root@uswuxsdbkp00:~# 
```
I think there are a few problems here:
* ZFS doesn't display the UID/GID when it can't resolve the name
* ZFS doesn't appear to have a flag to force returning the UID/GID instead of the name
* Maybe the docs are confusing, but there doesn't appear to be a way to say "remove all permissions from this dataset" (```zfs unallow -e tank/myfolder``` doesn't do it)

**Comment by behlendorf** [comment]
@darkpixel this was a fix for this in PR #10280.  It's included in the 2.0 releases, but hasn't been backported.

**Comment by darkpixel** [comment]
I'm not sure how I missed that bug when searching.  I must have forgotten to remove 'is:open' from the search.  Thanks @behlendorf!
